crates/core/component/auction/src/component/dutch_auction.rs

```diff
@@ -51,11 +64,183 @@ pub(crate) trait DutchAuctionManager: StateWrite {
+                        ?e,
+                        ?auction_lp_id,
+                        ?auction_id,
+                        "failed to close dutch auction LP"
+                    )
+                })
+                .expect("no state incoherence")
+        } else {
+            Balance::zero()
+        };
+
+        // We remove the execution trigger that we are currently processing:
+        self.unset_trigger_for_dutch_id(auction_id, trigger_height);
+
+        // Prepare a new auction, based on the previous one.
+        let mut new_dutch_auction = DutchAuction {
+            description: old_dutch_auction.description,
+            state: old_dutch_auction.state,
+        };
+
+        let balance_input_asset = lp_reserves
+            .provided()
+            .filter(|v| v.asset_id == input.asset_id)
+            .map(|v| v.amount)
+            .sum::<Amount>();
+        let balance_output_asset = lp_reserves
+            .provided()
+            .filter(|v| v.asset_id == output_id)
+            .map(|v| v.amount)
+            .sum::<Amount>();
+
+        new_dutch_auction.state.current_position = None;
+        // We overwrite the input amount
+        new_dutch_auction.state.input_reserves = balance_input_asset;
+        new_dutch_auction.state.output_reserves += balance_output_asset;
+        new_dutch_auction.state.next_trigger = 0;
+
+        // Compute the current step index, between 0 and `step_count`.
+        let step_index = auction_trigger
+            .compute_step_index(trigger_height)
+            .expect("trigger data is validated");
+
+        if step_index >= step_count {
+            // If the termination condition has been reached, we set the auction
+            // sequence to 1 (Closed).
+            new_dutch_auction.state.sequence = 1;
+        } else {
+            // Otherwise, we compute the next trigger height and generate a liquidity
+            // position for the new auction round.
+            let next_trigger = auction_trigger
+                .compute_next_trigger_height(trigger_height)
+                .expect("trigger data is validated")
+                .expect("the step count has not been reached");
+
+            // We compute the price parameters for the LP:
+            let (p, q) = compute_pq_at_step(&new_dutch_auction.description, step_index);
+
+            // Next, we want to construct an auction LP position and send it to the DEX.
+            // The nonce must be chosen so that the resulting position id is unique:
+            // `PositionManager::open_position` will reject duplicates.
+            //
+            // To do this, we keep track of our number of attempts at opening a position,
+            // and compute:
+            // position_nonce = H(auction_nonce || step_index || attempt_counter)
+            // until the resulting position id (based on the nonce) is unique and accepted
+            // by the DEX.
+            let mut attempt_counter = 0u64;
+            loop {
```

Comment:
structure suggestion: could this be broken out into an `allocate_position` method that returns the position?